src/pt/taosect/src/eu/kanade/tachiyomi/extension/pt/taosect/TaoSect.kt

```diff
@@ -0,0 +1,316 @@
+    override val baseUrl = "https://taosect.com/"
+    override val lang = "pt-BR"
+    override val supportsLatest = false
+
+    override val client by lazy {
+        network.cloudflareClient.newBuilder()
+            .rateLimit(2)
+            .build()
+    }
+
+    val readerUrl = "${baseUrl}leitor-online/"
+    val apiUrl = "${baseUrl}wp-admin/admin-ajax.php"
+
+    // ===== Popular =====
+    override fun popularMangaRequest(page: Int): Request {
+        return GET(readerUrl, headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+
+        val mangas = document.select(".container-obras-populares .manga_item").map { element ->
+            val imageContainer = element.select(".container_imagem")
+            val url = imageContainer.select(".link-img").attr("href")
+            val coverImage = REGEX_FOR_IMAGE_ON_STYLE.find(imageContainer.attr("style"))
+                ?.groupValues
+                ?.get(1)
+
+            MangaDto(
+                url = url.split(baseUrl)[1],
+                coverImage = coverImage,
+            )
+        }
+
+        return MangasPage(mangas.map { it.toSManga() }, false)
+    }
+
+    // ===== Latest =====
+    override fun latestUpdatesRequest(page: Int) = throw UnsupportedOperationException()
+
+    override fun latestUpdatesParse(response: Response) = throw UnsupportedOperationException()
+
+    // ===== Search =====
+    /**
+     * Is not possible to know if we will have a next page. This is a workflow to not get
+     * 'No Results found'.
+     */
+    private var lastResult: SManga? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val categoryId = filters.filterIsInstance<CategoryFilter>().firstOrNull()?.selected
+
+        if (page == 1) lastResult = null
+
+        return POST(
+            apiUrl,
+            headers,
+            buildSerializedQuery(
+                page = page,
+                query = query,
+                categoryId = categoryId?.toIntOrNull(),
+            ).toRequestBody("application/x-www-form-urlencoded; charset=UTF-8".toMediaType()),
+        )
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+
+        val mangas = document.select(".manga_item").map { element ->
+            val imageContainer = element.select(".container_imagem")
+            val contentContainer = element.select(".conteudo_manga_item")
+
+            val titleElement = contentContainer.select(".titulo_manga_item")
+
+            val url = titleElement.select("a").attr("href")
+            val coverImage = REGEX_FOR_IMAGE_ON_STYLE.find(imageContainer.attr("style"))
+                ?.groupValues
+                ?.get(1)
+
+            val description = contentContainer.select("p").text().trim()
+
+            MangaDto(
+                title = titleElement.text().trim(),
```

Comment:
`.text()` is already trimmed
```suggestion
                title = titleElement.text(),
```